### DISCUSSION TITLE
Don't regenerate pages for the same id, different slug

### DISCUSSION THREAD

**Discussion by lazandrei19** [vercel/next.js#17253] [Help]
I have this blog and the paths to the articles are `/article/[id]/[slug]`. When generating the fetching the data from the API, only the id is used. What this means is that if someone accesses the page using the wrong slug, it won't return a 404, instead it'll regenerate the page, consuming more disk space.
Is there any way to avoid this?

**Comment by lfades** [comment]
If you're using `getStaticProps` a 404 should be happening instead, because only the known paths that were generated at build time will be accessible by your visitors. If you're using SSR then make sure to 404 on paths that have an unexpected/unknown `slug`. And if you're not using it for anything then don't add a slug to avoid duplicated content.
